cg/apps/demultiplex/sample_sheet/api.py

```diff
@@ -0,0 +1,166 @@
+import logging
+import os
+from pathlib import Path
+
+import click
+from pydantic import ValidationError
+
+from cg.apps.demultiplex.sample_sheet.create import create_sample_sheet_content
+from cg.apps.demultiplex.sample_sheet.sample_models import FlowCellSample
+from cg.apps.demultiplex.sample_sheet.sample_sheet_validator import SampleSheetValidator
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.apps.lims import LimsAPI
+from cg.apps.lims.sample_sheet import get_flow_cell_samples
+from cg.constants.constants import FileFormat
+from cg.constants.demultiplexing import BclConverter
+from cg.exc import FlowCellError, HousekeeperFileMissingError, SampleSheetError
+from cg.io.controller import WriteFile, WriteStream
+from cg.meta.demultiplex.housekeeper_storage_functions import (
+    add_and_include_sample_sheet_path_to_housekeeper,
+)
+from cg.models.flow_cell.flow_cell import FlowCellDirectoryData
+
+LOG = logging.getLogger(__name__)
+
+
+class SampleSheetAPI:
+    """Sample Sheet API class."""
+
+    def __init__(self, flow_cell_dir: str, hk_api: HousekeeperAPI, lims_api: LimsAPI) -> None:
+        self.flow_cell_runs_dir = Path(flow_cell_dir)
+        self.hk_api: HousekeeperAPI = hk_api
+        self.lims_api: LimsAPI = lims_api
+        self.dry_run: bool = False
+        self.force: bool = False
+        self.validator = SampleSheetValidator()
```

Comment:
```suggestion
    def __init__(self, config: CGConfig) -> None:
        self.flow_cell_runs_dir = config.flow_cell_dir
        self.hk_api: HousekeeperAPI = config.houskeeper_api
        self.lims_api: LimsAPI = config.lims_api
        self.dry_run: bool = False
        self.force: bool = False
        self.validator = SampleSheetValidator()
```

Reply:
This API is defined inside the config, so I can't have it as parameter

Reply:
What api are you referrring to here?

> This API is defined inside the config, so I can't have it as parameter

Reply:
> What api are you referrring to here?
> 
> > This API is defined inside the config, so I can't have it as parameter

I am referring to SampleSheetAPI. I don't know, I think that it could work as you suggested, but I just used the same signature as the demultiplex API. Im not sure about the pros and cons of each approach

Reply:
Ah, I see. We typically pass the config in the CLI where these things are accessed, then from the config we get the APIs we need. One can either do that in the cli or pass on the config into the api where that stuff gets done in the init. 

I am not aware of any pros and cons; one thing that it impacts is reading comprehension.  I guess if you prefer having explicitly stated what apis are required when initialising then leave it as is. 

To me it would make sense to just pass on the config and let the rest be handled internally. Feel free to resolve :)

Reply:
Resolving for now, further discussion on this topic is encouraged in https://github.com/Clinical-Genomics/atlas/pull/2374